superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/buildQuery.ts

```diff
@@ -46,6 +47,11 @@ export default function buildQuery(formData: QueryFormData) {
     ...formData,
   };
 
+  if (!!baseFormData.orderby) {
+    baseFormData.x_axis_sort = formData.orderby.label;
```

Comment:
I think you may need to do this here to make sure we can handle all types of metrics (please also add the import):
```suggestion
    baseFormData.x_axis_sort = getMetricLabel(formData.orderby);
```